src/main/java/org/codehaus/plexus/classworlds/realm/ClassRealm.java

```diff
@@ -272,6 +272,30 @@ public void addURL( URL url )
         }
     }
 
+    // java11
+    protected Class<?> findClass(String moduleName, String name)
+    {
+        if (moduleName != null)
+        {
+            return null;
```

Comment:
does this work because on Maven plugins are not "modular jars" ?
or should be call super.findClass( moduleName, name ) ?

Reply:
first answer is likely yes and because ClassRealm ecosystem does not support modules actually

When it will be the case it can need some more love bu today I don't see a case it can be hit

Reply:
@rmannibucau 
I see what you are aiming for, but there is Maven compiler with Java 1.6.
Pls check the dependencies and maybe you will find some reflection tool. If not then you need to call the new signature `findClass( moduleName, name ) ` via Java Reflection, my doubt.

Second question, what if somebody, not in Maven, want's to use the `moduleName`. Is it regular way to `return null`?

Reply:
@Tibor17 hmm, calling it by reflection actually means you don't use ClassRealm but the URLClassLoader so at the end it is a case which does not really exist, right? Also it does not change much since the parent implementation is just findClass(name) if you check so i'm just bypassing a few steps to avoid a more complicated code.
Returning null is the expected default case when you don't really support module (which would requires some more advanced logic and integration with classes not available in java 6)

side note: the code does not compile on java 6 - before my change, it requires java 7 anyway so the pom is likely wrong ;).

Reply:
Do we have some Logger here ?
so we can log at a very low level (like DEBUG/TRACE) that we are hitting this case.

Reply:
I don't think so, didnt find one at least. JUL is an option but it can have negative impacts on redirections in maven if called too early